services/utils.py

```diff
@@ -19,6 +19,10 @@ def get_service_from_service_key(request, raise_exception: bool = True) -> Servi
 
         # get_from_key also checks that the key is still valid
         service_key = ServiceAPIKey.objects.get_from_key(key)
+
+        if service_key.has_expired:
```

Comment:
Nice :+1:

Reply:
Though, shouldn't the ServiceAPIKey.objects.get_from_key(key) already check for validity?

Reply:
I thought so too, but internally the `get_from_key` calls the model's `is_valid` (and not the manager's as I originally thought), and that only ckecks if the key itself is valid, it doesn't include the expiration

```python
def is_valid(self, key: str) -> bool:
    return type(self).objects.key_generator.verify(key, self.hashed_key)
```